### PR TITLE
Add lease to runs.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR adds a PowerShell script that adds a retention lease to a pipeline run. Because you can only have one lease it first checks to see if a lease exists already, and if it does it removes it and adds it again for the specified validity period (there doesn't appear to be a patch operation according to the REST API docs).

### CODE REVIEW

**Comment by mitchdenny** [comment]
Closes: https://github.com/Azure/azure-sdk/issues/1059

File: eng/common/scripts/Add-RetentionLease.ps1 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+$existingLeases = Get-RetentionLeases -Organization $Organization -Project $Project -DefinitionId $DefinitionId -RunId $RunId -OwnerId $OwnerId -Base64EncodedAuthToken $Base64EncodedAuthToken
+
+if ($existingLeases.count -ne 0) {
+    LogDebug "Found $($existingLeases.count) leases, will delete them first."
+
+    foreach ($lease in $existingLeases.value) {
+        LogDebug "Deleting lease: $($lease.leaseId)"
+        Delete-RetentionLease -Organization $Organization -Project $Project -LeaseId $lease.leaseId -Base64EncodedAuthToken $Base64EncodedAuthToken
+    }
+
```
Comment (weshaggard) [inline eng/common/scripts/Add-RetentionLease.ps1:37]: nit: empty line

**Review by weshaggard — APPROVED** [review]
Looks reasonable.

File: eng/common/pipelines/templates/steps/retain-run.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    inputs:
+      pwsh: true
+      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Add-RetentionLease.ps1
+      arguments: >
+        -Organization azure-sdk
+        -Project $(System.TeamProject)
+        -DefinitionId $(System.DefinitionId)
+        -RunId $(Build.BuildId)
+        -OwnerId Pipeline
+        -DaysValid 365
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/retain-run.yml:14]: We might want this to be a template parameter.

**Pull request merged**